### DISCUSSION TITLE
Export `resource-query` for use by other projects?

### DISCUSSION THREAD

**Discussion by SteVwonder** [flux-framework/flux-sched#869] [Ideas]
For projects like `flux-coral2`, it could be useful to have access to `resource-query` for testing that certain jobspecs will work on a given resource file for a given match policy.  We could always put some of those tests into the flux-sched repo (like we did with the MTL tests), but if you are dynamically generating a JGF from a DWS API and want to test that JGF with `resource-query`, that is harder to do with that workaround.  At the same time, it probably doesn't make sense to have a `flux resource-query` command exposed to users.  I'm curious what other projects do in this scenario.

**Comment by vsoch** [comment]
We have [fluxion-go](https://github.com/flux-framework/fluxion-go) bindings that can partially address this now, and several examples of using them for (what comes down to) small applications that are similar to resource query, although they aren't currently interactive (e.g., with the terminal input).

- https://github.com/converged-computing/flex-ice-cream
- https://github.com/converged-computing/flex-aws-topology
- https://github.com/converged-computing/flex-archspec

These are actually more generic use cases of the same design - not even for resource graphs, but other things (ice cream, topology, and archspec, respectively).
